plugins/labhub.py

```diff
@@ -1,7 +1,262 @@
+                             filter(lambda x: (x.full_name.split('/')[0] ==
+                                               self.GH_ORG_NAME),
+                                    self.IGH.write_repositories)}
+            self.gl_repos = {repo.full_name.split('/')[1]: repo for repo in
+                             filter(lambda x: (x.full_name.split('/')[0] ==
+                                               self.GL_ORG_NAME),
+                                    self.IGL.write_repositories)}
+            self.REPOS = {**self.gh_repos, **self.gl_repos}
+        except RuntimeError:
+            self.log.error('Either of GH_TOKEN or GL_TOKEN is not set')
+
+    @classmethod
+    def get_teams(cls, token, org_name):
+        """
+        Return a dict of team names as keys and corresponding
+        ``github3.orgs.Team`` objects as values.
+
+        :param token:    Personal access token or oauth token.
+        :param org_name: Name of the organization.
+        """
+        cls.GH = github3.login(token=token)
+        assert cls.GH is not None
+        cls.GH_ORG = cls.GH.organization(org_name)
+        teams = dict()
+        for team in cls.GH_ORG.iter_teams():
+            teams[team.name] = team
+        return teams
+
+    @re_botcmd(pattern=r'(?:(?:invite)|(?:inv))\s+(\w+)\s*(?:to)\s+(\w+)')
+    def invite_cmd(self, msg, match):
+        """
+        Invite given user to given team. By default it invites to
+        "newcomers" team.
+        """
+        if not hasattr(self, 'TEAMS'):
+            self.TEAMS = LabHub.get_teams(os.environ.get('GH_TOKEN'),
+                                          self.GH_ORG_NAME)
+
+        invitee = match.group(1)
+        inviter = msg.frm.nick
+
+        team = (self.GH_ORG_NAME + ' newcomers' if match.group(2) is None
+                else match.group(2))
+
+        self.log.info('{} invited {} to {}'.format(inviter, invitee, team))
+
+        if self.TEAMS[self.GH_ORG_NAME + ' maintainers'].is_member(invitee):
+            valid_teams = ['newcomers', 'developers', 'maintainers']
+            if team.lower() not in valid_teams:
+                return 'Please select from one of the ' + ', '.join(valid_teams)
+            team_mapping = {
+                'newcomers': self.GH_ORG_NAME + ' newcomers',
+                'developers': self.GH_ORG_NAME + ' developers',
+                'maintainers': self.GH_ORG_NAME + ' maintainers'
+            }
+
+            # send the invite
+            self.TEAMS[team_mapping[team.lower()]].invite(invitee)
+            return self.INVITE_SUCCESS[team.lower()].format(invitee)
+        else:
+            return ('@{}, you are not a maintainer, only maintainers can invite'
+                    ' other people. Nice try :poop:'.format(inviter))
+
+    def callback_message(self, msg):
+        """Invite the user whose message includes the holy 'hello world'"""
+        if not hasattr(self, 'TEAMS'):
```

Comment:
create a instance `@property` `teams`, which abstracts away the dynamic building of the data when required